Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/Events/EventReceivers/SpeechKeywordReceiver.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using Microsoft.MixedReality.Toolkit.Input;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// The internal event receiver for the events defined in the SpeechKeyword Interaction Event Configuration.
+    /// </summary>
+    public class SpeechKeywordReceiver : BaseEventReceiver
+    {
+        public SpeechKeywordReceiver(BaseInteractionEventConfiguration eventConfiguration) : base(eventConfiguration) { }
+
+        private SpeechKeywordEvents SpeechKeywordEventConfig => EventConfiguration as SpeechKeywordEvents;
+
+        private SpeechInteractionEvent onSpeechKeywordRecognized => SpeechKeywordEventConfig.OnAnySpeechKeywordRecognized;
+        
+        private List<KeywordEvent> keywordsAndResponses => SpeechKeywordEventConfig.Keywords;
+
+        /// <inheritdoc />
+        public override void OnUpdate(StateManager stateManager, BaseEventData eventData)
+        {
+            bool keywordRecognized = stateManager.GetState(StateName).Value > 0;
+
+            if (keywordRecognized)
+            {
+                SpeechEventData speechData = eventData as SpeechEventData;
+
+                onSpeechKeywordRecognized.Invoke(speechData);
+
+                bool speechKeywordRecognized = speechData.Command.Keyword != null;
+
+                if (speechKeywordRecognized)
+                {
+                    // Get the keyword that was recognized
+                    string speechEventKeyword = speechData.Command.Keyword;
+
+                    // Capitalize the first letter of the state in case the keyword is in lower case
```

Comment:
Capitalizing first letter like this and doing case-sensitive string compare feels like it could break in the future. 
 Instead, consider using a case-insensitive string comparison rather than capitalizing first letter, which could lead to cases where the registered keyword's capitalization doesn't match the keyword that comes back from MRTK (e.g. all of a sudden MRTK starts sending back keywords recognized in all caps).

Reply:
Added case insensitive string comparison.